addons/mail/static/src/core_ui/message.js

```diff
@@ -315,15 +316,22 @@ export class Message extends Component {
     }
 
     onClickPin() {
-        const prompt = this.message.pinned_at
-            ? _t("Are you sure you want to remove this pinned message?")
-            : _t(
-                  "The following message will be pinned to the channel. Are you sure you want to continue?"
-              );
+        const thread = this.message.originThread;
         this.env.services.dialog.add(MessageConfirmDialog, {
+            confirmColor: this.message.pinned_at ? "btn-danger" : undefined,
+            confirmText: this.message.pinned_at ? _t("Yes, remove it please") : _t("Yeah, pin it!"),
             message: this.message,
             messageComponent: Message,
-            prompt,
+            prompt: this.message.pinned_at
+                ? _t("Well nothing lasts forever, but are you sure you want to unpin this message?")
+                : sprintf(
+                      _t(
+                          "You sure want this message pinned to %(conversation_name)s forever and ever?"
+                      ),
+                      { conversation_name: thread.prefix + thread.displayName }
```

Comment:
I know you prefer short names 😺
```suggestion
                      _t(
                          "You sure want this message pinned to %(conversation)s forever and ever?"
                      ),
                      { conversation: thread.prefix + thread.displayName }
```